File: 59-Spiral Matrix II-螺旋矩阵 II/59-Spiral Matrix II-螺旋矩阵 II.js
/**
 * @param {number} n
 * @return {number[][]}
 */
 var generateMatrix = function(n) {
    let top = 0
    let bottom = n - 1
    let left = 0
    let right = n - 1
    let num = 1
    let res = []
    for(let i = 0; i < n; i++){
        res[i] = []
    }
    while(num <= n*n){
        for(let i = left; i <= right; i++, num++) res[top][i] = num
        top++
        for(let i = top; i <= bottom; i++, num++) res[i][right] = num
        right--
        for(let i = right; i >= left; i--, num++) res[bottom][i] = num
        bottom--
        for(let i = bottom; i >= top; i--, num++) res[i][left] = num
        left++
    }
    return res
};

File: 59-Spiral Matrix II-螺旋矩阵 II/59-Spiral Matrix II-螺旋矩阵 II.java
class Solution {
    public int[][] generateMatrix(int n) {
        int top = 0;
        int bottom = n - 1;
        int left = 0;
        int right = n - 1;
        int num = 1;
        int[][] res = new int[n][n];
        while(num <= n*n){
            for(int i = left; i <= right; i++, num++) res[top][i] = num;
            top++;
            for(int i = top; i <= bottom; i++, num++) res[i][right] = num;
            right--;
            for(int i = right; i >= left; i--, num++) res[bottom][i] = num;
            bottom--;
            for(int i = bottom; i >= top; i--, num++) res[i][left] = num; 
            left++;
        }
        return res;
    }
}